crates/storage/provider/src/providers/blockchain_provider.rs

```diff
@@ -1503,19 +1504,49 @@ mod tests {
         database_blocks: usize,
         in_memory_blocks: usize,
         block_range_params: BlockRangeParams,
+        with_random_txs: bool,
     ) -> eyre::Result<(
         BlockchainProvider2<Arc<TempDatabase<DatabaseEnv>>>,
         Vec<SealedBlock>,
         Vec<SealedBlock>,
         Vec<Vec<Receipt>>,
     )> {
-        let (database_blocks, in_memory_blocks) = random_blocks(
+        let (mut database_blocks, mut in_memory_blocks) = random_blocks(
```

Comment:
no need for another separate function, we should instead pass the `block_range_params.tx_count` here, and then further down into `random_block_range`. That way, we can just set the `tx_count` field of `BlockRangeParams` in tx-related tests

Reply:
Oh yes cleaner, just fixed